lib/streamlit/elements/arrow_altair.py

```diff
@@ -286,13 +343,134 @@ def _is_date_column(df: pd.DataFrame, name: str) -> bool:
     column = df[name]
     if column.size == 0:
         return False
+
     return isinstance(column[0], date)
 
 
+def _maybe_melt(
```

Comment:
maybe add a docstring what this is supposed to do?

Reply:
Did a small refactoring on this and added a docstring 👍